pkg/store/proxy.go

```diff
@@ -178,10 +180,10 @@ func (s *ProxyStore) Series(r *storepb.SeriesRequest, srv storepb.Store_SeriesSe
 			storeDebugMsgs = append(storeDebugMsgs, fmt.Sprintf("store %s queried", st))
 
 			// This is used to cancel this stream when one operations takes too long.
-			seriesCtx, closeSeries := context.WithCancel(gctx)
-			defer closeSeries()
+			storeCtx, cancelStore := context.WithCancel(queryCtx)
+			defer cancelStore()
```

Comment:
We already defer this which should cancel all so wg.Wait should not block forever unless there is a bug inside streamSeriesSet